pkg/operator/ceph/cluster/controller.go

```diff
@@ -808,10 +810,52 @@ func (c *ClusterController) handleDelete(cluster *cephv1.CephCluster, retryInter
 			cluster.Namespace, retryInterval)
 		<-time.After(retryInterval)
 	}
+	return nil
+}
+
+func (c *ClusterController) handleCSIPVAttachment(cluster *cephv1.CephCluster, retryInterval time.Duration) error {
+	retryCount := 0
+	drivers := []string{csi.CephFSDriverName, csi.RBDDriverName}
+	for {
+		logger.Infof("checking any PVC created by drivers %v and %v with clusterID %v", csi.CephFSDriverName, csi.RBDDriverName, cluster.Namespace)
+		// check any PV is created in this cluster
+		attachmentsExist, err := k8sutil.CheckPVPresentInCluster(c.context.Clientset, drivers, cluster.Namespace)
+
+		if err != nil {
+			err = fmt.Errorf("failed to list PersistentVolumes %+v", err)
+			logger.Error(err)
+			return err
+		}
+		// no PVC created in this cluster
+		if !attachmentsExist {
+			logger.Infof("no volume attachments for cluster %s", cluster.Namespace)
+			break
+		}
 
+		retryCount++
+		if retryCount == clusterDeleteMaxRetries {
+			logger.Warningf(
+				"exceeded retry count while waiting for volume attachments for cluster %s to be cleaned up", cluster.Namespace)
+			break
+		}
+
+		logger.Infof("waiting for volume attachments in cluster %s to be cleaned up. Retrying in %v",
+			cluster.Namespace, retryInterval)
+		<-time.After(retryInterval)
+	}
 	return nil
 }
 
+func (c *ClusterController) handleDelete(cluster *cephv1.CephCluster, retryInterval time.Duration) error {
+	var err error
+	if csi.CSIEnabled() {
+		err = c.handleCSIPVAttachment(cluster, retryInterval)
+		return err
+	}
+	err = c.handleFexVolAttachment(cluster, retryInterval)
```

Comment:
how about moving the check for `os.Getenv(enableFlexDriver) != "false"` here so it follows the same pattern as checking if CSI is enabled before continuing?